### PR TITLE
Removed XZ compression feature of serialized IValues

### LINKED ISSUE #252: Drop xz-java dependency
Considering recent news, we're trying to take a close peek at our dependencies, and the support behind them.

Now we actually depend on xz-java, which based on most analysis [doesn't appear to have received meaningful commits from the harmful actor](https://git.tukaani.org/?p=xz-java.git;a=shortlog). And indeed, [only 1 commit that changes code](https://git.tukaani.org/?p=xz-java.git;a=commit;h=8e46fdf903563b75849a37150be1cbf51ba12c53), and I've reviewed it, it looks fine. So I have no specific concerns with regards to xz-java in vallang.

However, I did lead me to think about, why do we have it in vallang? It's only used in the binary serializer, in a compression mode that was up until very recently not working. I don't think there exists any files that use that compression mode. Especially since there has been no release of vallang with a working `EXTREME` level of compression.

I think we could remove the xz-java dependency, and stop support for that extreme compression mode, without hurting any user. And throw an exception in the reader to use an older version to downgrade the file to a lower level of compresssion (which I think nobody will have ever hit).

### PR DESCRIPTION
**Pull request by DavyLandman**
Fixes #252

### CODE REVIEW

File: src/main/java/io/usethesource/vallang/io/binary/stream/Compressor.java [hunk 1]
```diff
@@ -78,7 +75,7 @@ public static InputStream wrapStream(InputStream raw, int algorithm) throws IOEx
             case Header.Compression.GZIP:
                 return new GZIPInputStream(raw);
             case Header.Compression.XZ:
-                return new XZInputStream(raw);
+                throw new IOException("XZ compression is not supported anymore, please use an older version of rascal/vallang to open the file and store it with a different level of compression");
```
Comment (jurgenvinju) [inline src/main/java/io/usethesource/vallang/io/binary/stream/Compressor.java:78]: same same

File: src/main/java/io/usethesource/vallang/io/binary/stream/Header.java [hunk 2]
```diff
@@ -17,6 +17,7 @@
     public static final class Compression {
         public static final byte NONE = 0;
         public static final byte GZIP = 1;
+        /** Not used anymore, it was never used, and we're cutting down on dependencies */
```
Comment (jurgenvinju) [inline src/main/java/io/usethesource/vallang/io/binary/stream/Header.java:20]: rename `XZ` to `USED_TO_BE_XZ` or something?

File: src/main/java/io/usethesource/vallang/io/binary/stream/IValueInputStream.java [hunk 3]
```diff
@@ -49,6 +50,9 @@ public IValueInputStream(InputStream in, IValueFactory vf, Supplier<TypeStore> t
         while (read < currentHeader.length) {
             read += in.read(currentHeader, read, currentHeader.length - read);
         }
+        if (!Arrays.equals(currentHeader, Header.MAIN)) {
+            throw new IOException("Incorrect file header, are you sure this is a valid file containing a serialized value?");
```
Comment (jurgenvinju) [inline src/main/java/io/usethesource/vallang/io/binary/stream/IValueInputStream.java:54]: pedantic error message (a rhetorical question). "Unexpected file header: "; perhaps print the current header and the expected header inside the message?

**Review by jurgenvinju — CHANGES_REQUESTED** [review]
nice to get this over with!

File: src/main/java/io/usethesource/vallang/io/binary/stream/IValueOutputStream.java [hunk 4]
```diff
@@ -55,7 +55,7 @@ public enum CompressionRate {
         Light(Header.Compression.ZSTD, 1),
         Normal(Header.Compression.ZSTD, 5),
         Strong(Header.Compression.ZSTD, 13),
-        Extreme(Header.Compression.XZ, 6),
+        Extreme(Header.Compression.ZSTD, 19),
```
[the remark this answers is not in the document]
Comment (DavyLandman) [reply src/main/java/io/usethesource/vallang/io/binary/stream/IValueOutputStream.java:58]: Well, the zstd compression modes are different than xz. and 19 is the highest for zstd that doesn't require special flags. and it's "close" to xz at 6.

**Comment by DavyLandman** [comment]
@jurgenvinju thanks, done, squash & merge when approved :)

**Review by jurgenvinju — APPROVED** [review]

**Comment by jurgenvinju** [comment]
Thanks!

**Pull request merged**